ui/src/components/annotation/shape-factory.component.tsx

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Shape } from '../shapes/interfaces';
+import { Polygon } from '../shapes/polygon.component';
+import { Rectangle } from '../shapes/rectangle.component';
+
+interface ShapeFactoryProps {
+    shape: Shape;
+    styles: React.SVGProps<SVGPolygonElement & SVGRectElement>;
+    ariaLabel: string;
+}
+export const ShapeFactory = ({ shape, styles, ariaLabel }: ShapeFactoryProps) => {
+    switch (shape.shapeType) {
```

Comment:
Do we need a default case here? Since there are only two possible cases, perhaps a straightforward `if/else` would be sufficient

Reply:
Updated